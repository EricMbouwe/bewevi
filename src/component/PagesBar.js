import PropTypes from 'prop-types';

function PagesBar({
  pages, handleClick, selfPage,
}) {
  return (
    <div className="pages-bar main">
      {pages.map(page => (
        <button
          type="button"
          key={page[0]}
          onClick={() => handleClick(page[0])}
          className={`pages-bar page-button ${selfPage === page[0] ? 'current-page' : ''}`}
        >
          {page[1]}
        </button>
      ))}
    </div>
  );
}
PagesBar.propTypes = {
  pages: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
    ).isRequired,
  ).isRequired,
  handleClick: PropTypes.func.isRequired,
  selfPage: PropTypes.number.isRequired,

};
export default PagesBar;
