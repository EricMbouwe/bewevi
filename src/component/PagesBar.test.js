import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import PagesBar from './PagesBar';

const pages = [
  [3, 'Previous'],
  [4, '4'],
  [5, 'Next'],
];
const handleClick = jest.fn().mockImplementation(className => className + className);
const selfPage = 4;
describe('<PagesBar />', () => {
  it('contains expected texts', () => {
    render(
      <PagesBar
        pages={pages}
        handleClick={handleClick}
        selfPage={selfPage}
      />,
    );

    expect(screen.getByText(/Previous/i)).toBeInTheDocument();
    expect(screen.getByText(/4/i)).toBeInTheDocument();
    expect(screen.getByText(/Next/i)).toBeInTheDocument();
  });

  it('is triggering handleClick correctly', () => {
    render(
      <PagesBar
        pages={pages}
        handleClick={handleClick}
        selfPage={selfPage}
      />,
    );

    userEvent.click(screen.getByText(/Previous/i));
    expect(handleClick.mock.calls.length).toBe(1);
    expect(handleClick.mock.calls[0][0]).toBe(3);

    userEvent.click(screen.getByText(/4/i));
    expect(handleClick.mock.calls.length).toBe(2);
    expect(handleClick.mock.calls[1][0]).toBe(4);

    userEvent.click(screen.getByText(/Next/i));
    expect(handleClick.mock.calls.length).toBe(3);
    expect(handleClick.mock.calls[2][0]).toBe(5);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <PagesBar
        pages={pages}
        handleClick={handleClick}
        selfPage={selfPage}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
