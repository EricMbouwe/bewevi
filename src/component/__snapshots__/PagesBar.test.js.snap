// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PagesBar /> renders correctly 1`] = `
<div
  className="pages-bar main"
>
  <button
    className="pages-bar page-button "
    onClick={[Function]}
    type="button"
  >
    Previous
  </button>
  <button
    className="pages-bar page-button current-page"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="pages-bar page-button "
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
